devwilliamy/coverland
# Implemented Taxjar and Added Payment Processing Messages

Tax will now be calculated via Taxjar
Tax is calculated after shipping is decided
Once tax is calculated it has to update the following
- Orders Table (total amount and tax amount)
- Order Items Table (tax amount)
- Order Tax Table (tax breakdown)
- Stripe PaymentIntent (amount and metadata)
- Paypal Total Price
- Email contains tax now

Extra stuff:
- Renamed totalMSRPPrice to orderTotal
- Renamed getTotalPrice to getCartTotalPrice
- Updated conversion to include tax except for meta / fb
  - decided not to for now because they don't have a slot for tax 
- Created payment processing messages for stripe/paypal

Order Breakdown:
![image](https://github.com/user-attachments/assets/6f7a4ec7-92db-448c-82ee-214468fed1d9)

Paypal Confirmation:
![image](https://github.com/user-attachments/assets/3aa433ac-4b64-4ade-9d5d-c34d193f608a)

Stripe Confirmation:
![image](https://github.com/user-attachments/assets/7d8d2e6b-5b0b-4b19-aeba-b0c19b2bd7dd)

Payment Processing
![image](https://github.com/user-attachments/assets/2d7fd945-906e-4700-b70b-e689b2f82bb5)

![image](https://github.com/user-attachments/assets/1a24b355-d281-4110-9cd9-5b74883e44b9)

### How To Test
- Add an item to cart
- Go to checkout
- After adding address & selecting shipping, estimated tax should change (for California only)
- Make sure order total is updated properly and adds up correctly
- Checkout with both Stripe / Paypal, ensure the price is updated on their side
- Verify tax breakdown & updated order total are reflected in Stripe, Paypal, email, and admin panel (order history not updated yet)
- Also try checking out with a state other than California, and make sure you can finish the transaction (I ran into an issue and couldn't continue but I fixed it, just need to see if there are more)
